### ISSUE TITLE
Add ability to queue only without sending

### ISSUE THREAD

**Issue by icemac** [zopefoundation/Products.MailHost#14]
For a staging system it would be nice to be able to configure that mails are only queued but not sent. Currently the mail sending thread is started each time a mail is sent.

https://github.com/zopefoundation/Products.MailHost/blob/cf275046a177ed49d5ee79b8bdd95f835d53be78/src/Products/MailHost/MailHost.py#L313-L322

I'd suggest to add a feature to disable starting of the thread during `_send` using an environment variable.
Do you think this is okay?

**Comment by jugmac00** [comment]
Unfortunately, this patch was not enough to prevent sending emails on a staging environment, as the above mentioned _send method is not the only place which calls `_startQueueProcessorThread`

The "queue processor thread" also gets started when you hit the "Save changes" button in the ZMI
https://github.com/zopefoundation/Products.MailHost/blob/4bc4b95486e935edd7392c1800859871abce6962/src/Products/MailHost/MailHost.py#L164-L170

So, instead of trying to find all calls of `_startQueueProcessorThread` and prevent those being called, what about modifying the method declaration of `_startQueueProcessorThread` itself? And check within the method whether it should start a thread or not.

Something like...
```
    @synchronized(lock)
    def _startQueueProcessorThread(self):
        """ Start thread for processing the mail queue.
        """
        if not os.environ.get('MAILHOST_QUEUE_ONLY', False):
            key = self._getThreadKey()
            if key not in queue_threads:
                thread = QueueProcessorThread()
                thread.setMailer(self._makeMailer())
                thread.setQueuePath(self.smtp_queue_directory)
                thread.start()
                queue_threads[key] = thread
                LOG.info('Thread for %s started' % key)
```

**Comment by jugmac00** [comment]
Rethinking again... why does the "queue processor thread" get started when you hit the "save button" at all? This is not something I would expect, especially not as the "Start ..." link/button is no 2 cm below the save button.

So alternatively, or even additionally remove the "start queue processor thread" from the `manage_makeChanges` method completely.

**Comment by icemac** [comment]
I think it is enough not to start the thread on configuration changes.

**Comment by jugmac00** [comment]
@dataflake What's your take on this?

**Comment by dataflake** [comment]
I have no opinion on this.

**Comment by jugmac00** [comment]
Fixed by #20.

**Comment by jugmac00** [comment]
Fixed by #20
